godot-core/src/obj/gd.rs

```diff
@@ -338,15 +337,15 @@ impl<T: GodotClass> Gd<T> {
         })
     }
 
-    // Temporary workaround for bug in Godot that makes casts always succeed.
-    // (See https://github.com/godot-rust/gdext/issues/158)
-    // TODO(#234) remove this code once the bug is fixed upstream.
+    // See use-site for explanation.
+    // #[cfg(gdextension_api = "4.0")]
```

Comment:
is this cfg (and the one below) supposed to be a comment?

Reply:
I wanted to fix https://github.com/godot-rust/gdext/issues/158#issuecomment-1631340769 first, but it looks like it doesn't just work in Godot 4.1.

Maybe it's better to keep that for another PR and remove the `#[cfg]` entirely.